Tried to build for Qt6 using your branch with USE_GUI_CONFIG set to OFF, but
```
$ make
[  2%] Automatic MOC for target skulpture
[  2%] Built target skulpture_autogen
[  5%] Building CXX object src/CMakeFiles/skulpture.dir/skulpture_autogen/mocs_compilation.cpp.o
[  8%] Building CXX object src/CMakeFiles/skulpture.dir/sk_animations.cpp.o
[ 11%] Building CXX object src/CMakeFiles/skulpture.dir/sk_arrows.cpp.o
[ 14%] Building CXX object src/CMakeFiles/skulpture.dir/sk_buttons.cpp.o
[ 17%] Building CXX object src/CMakeFiles/skulpture.dir/sk_cache.cpp.o
[ 20%] Building CXX object src/CMakeFiles/skulpture.dir/sk_color.cpp.o
[ 23%] Building CXX object src/CMakeFiles/skulpture.dir/sk_combobox.cpp.o
[ 26%] Building CXX object src/CMakeFiles/skulpture.dir/sk_complex.cpp.o
[ 29%] Building CXX object src/CMakeFiles/skulpture.dir/sk_dial.cpp.o
[ 32%] Building CXX object src/CMakeFiles/skulpture.dir/sk_dock.cpp.o
[ 35%] Building CXX object src/CMakeFiles/skulpture.dir/sk_factory.cpp.o
[ 38%] Building CXX object src/CMakeFiles/skulpture.dir/sk_frames.cpp.o
[ 41%] Building CXX object src/CMakeFiles/skulpture.dir/sk_gradients.cpp.o
[ 44%] Building CXX object src/CMakeFiles/skulpture.dir/sk_header.cpp.o
[ 47%] Building CXX object src/CMakeFiles/skulpture.dir/sk_icons.cpp.o
[ 50%] Building CXX object src/CMakeFiles/skulpture.dir/sk_layout.cpp.o
/home/[REDACTED]/src/test/skulpture/src/sk_layout.cpp: In member function ‘virtual int SkulptureStyle::pixelMetric(QStyle::PixelMetric, const QStyleOption*, const QWidget*) const’:
/home/[REDACTED]/src/test/skulpture/src/sk_layout.cpp:212:14: error: ‘PM_DefaultTopLevelMargin’ was not declared in this scope
  212 |         case PM_DefaultTopLevelMargin: {
      |              ^~~~~~~~~~~~~~~~~~~~~~~~
/home/[REDACTED]/src/test/skulpture/src/sk_layout.cpp:218:14: error: ‘PM_DefaultChildMargin’ was not declared in this scope
  218 |         case PM_DefaultChildMargin: {
      |              ^~~~~~~~~~~~~~~~~~~~~
/home/[REDACTED]/src/test/skulpture/src/sk_layout.cpp:224:14: error: ‘PM_DefaultLayoutSpacing’ was not declared in this scope
  224 |         case PM_DefaultLayoutSpacing: {
      |              ^~~~~~~~~~~~~~~~~~~~~~~
/home/[REDACTED]/src/test/skulpture/src/sk_layout.cpp:283:42: error: ‘PM_DefaultChildMargin’ is not a member of ‘QStyle’
  283 |             PixelMetric metric = QStyle::PM_DefaultChildMargin;
      |                                          ^~~~~~~~~~~~~~~~~~~~~
/home/[REDACTED]/src/test/skulpture/src/sk_layout.cpp:285:34: error: ‘PM_DefaultTopLevelMargin’ is not a member of ‘QStyle’
  285 |                 metric = QStyle::PM_DefaultTopLevelMargin;
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [src/CMakeFiles/skulpture.dir/build.make:286: src/CMakeFiles/skulpture.dir/sk_layout.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:99: src/CMakeFiles/skulpture.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -62,4 +62,3 @@ endif()
 
 
 install(TARGETS ${LIBRARY_NAME} DESTINATION "${QT_PLUGINS_DIR}/styles")
-

--- a/src/config/CMakeLists.txt
+++ b/src/config/CMakeLists.txt
@@ -40,3 +40,4 @@ target_link_libraries(kstyle_skulpture_config
 
 install(TARGETS kstyle_skulpture_config DESTINATION ${QT_PLUGINS_DIR})
 install(FILES skulptureui.rc DESTINATION ${KDE_INSTALL_KXMLGUI5DIR}/skulpture)
+install(FILES ../skulpture.themerc DESTINATION ${KDE_INSTALL_DATADIR}/kstyle/themes)

--- a/src/config/sk_config.cpp
+++ b/src/config/sk_config.cpp
@@ -190,7 +190,7 @@ void SkulptureStyleConfig::updatePreview()
 			QString absFileName = tempDir.absoluteFilePath(fileName);
 			{
 				QSettings s(absFileName, QSettings::IniFormat);
-                                configManager->save(s);
+				configManager->save(s);
 				// make visible in other process
 				s.sync();
 			}
@@ -206,8 +206,12 @@ void SkulptureStyleConfig::updatePreview()
 
 			tempDir.remove(fileName);
 
-                        int margin = style->pixelMetric(QStyle::PM_DefaultTopLevelMargin);
-                        dialogLayout->setContentsMargins(margin, margin, margin, margin);
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 0, 0))
+			int margin = style->pixelMetric(QStyle::PM_LayoutLeftMargin, nullptr, this->window());
+#else
+			int margin = style->pixelMetric(QStyle::PM_DefaultTopLevelMargin);
+#endif
+			dialogLayout->setContentsMargins(margin, margin, margin, margin);
 			QList<QMdiSubWindow *> windows = mdiArea->findChildren<QMdiSubWindow *>();
             Q_FOREACH (QMdiSubWindow *window, windows) {
                                 window->setFocusPolicy(Qt::FocusPolicy(window->focusPolicy() & ~Qt::TabFocus));

--- a/src/sk_layout.cpp
+++ b/src/sk_layout.cpp
@@ -209,6 +209,7 @@ int SkulptureStyle::pixelMetric(PixelMetric metric, const QStyleOption *option, 
 
         case PM_SpinBoxSliderHeight: return 2;
 
+#if (QT_VERSION < QT_VERSION_CHECK(5, 0, 0))
         case PM_DefaultTopLevelMargin: {
             if (d->dialogMargins >= 0) {
                 return d->dialogMargins;
@@ -227,6 +228,7 @@ int SkulptureStyle::pixelMetric(PixelMetric metric, const QStyleOption *option, 
             }
             return ((fontHeight(option, widget) * 3) >> 3);
         }
+#endif
 
         case PM_ToolBarIconSize:
             return QCommonStyle::pixelMetric(metric, option, widget);
@@ -280,11 +282,22 @@ int SkulptureStyle::pixelMetric(PixelMetric metric, const QStyleOption *option, 
         case PM_LayoutTopMargin:
         case PM_LayoutRightMargin:
         case PM_LayoutBottomMargin: {
+#if (QT_VERSION < QT_VERSION_CHECK(5, 0, 0))
             PixelMetric metric = QStyle::PM_DefaultChildMargin;
             if ((option && (option->state & QStyle::State_Window)) || (widget && widget->isWindow())) {
                 metric = QStyle::PM_DefaultTopLevelMargin;
             }
             return pixelMetric(metric, option, widget);
+#else
+            int margins = d->widgetMargins;
+            if ((option && (option->state & QStyle::State_Window)) || (widget && widget->isWindow())) {
+                margins = d->dialogMargins;
+            }
+            if (margins >= 0) {
+                return margins;
+            }
+            return (fontHeight(option, widget) >> 1);
+#endif
         }
 
         case PM_LayoutHorizontalSpacing:

